hkmc2/shared/src/main/scala/hkmc2/codegen/Lifter.scala

```diff
@@ -613,6 +613,12 @@ class Lifter(handlerPaths: Opt[HandlerPaths])(using State, Raise):
         case Assign(t: TermSymbol, rhs, rest) if t.owner.isDefined =>
           ctx.getIsymPath(t.owner.get) match
             case Some(value) if !belongsToCtor(t.owner.get) =>
+              if (t.k is syntax.LetBind) && !t.owner.forall(_.isInstanceOf[semantics.TopLevelSymbol]) then
+                // TODO: can we do better?
+                raise(ErrorReport(
```

Comment:
I think we mentioned before that `ErrorReport` is reserved for errors in the user's code. This should be `InternalError`.

Reply:
Should InternalError not crash the compiler? Right now it seems it stop the codegen entirely if InternalError is raised.

Reply:
Whoops. `InternalError` _should_ crash the compiler; it is raised when the compiler finds it has a bug, for things that should never happen (failed assumptions). That's why it is purposely not handled in the diff tests.

This should be an `ErrorReport`. Though one could make a case that a separate kind of errors could be made to represent current compiler limitations (rather than user errors).

Reply:
There is no problem of crashing the compiler here anyway. The generated code would be wrong for private variables.